internal/services/automation/automation_job_schedule_resource.go

```diff
@@ -185,41 +176,53 @@ func resourceAutomationJobScheduleRead(d *pluginsdk.ResourceData, meta interface
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := jobschedule.ParseJobScheduleID(d.Id())
+	// the jobSchedule ID may be updated by Runbook, so need to get the real id by list API
+	tfID, err := parse.AutomationJobScheduleID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, *id)
+	js, err := GetJobScheduleFromTFID(ctx, client, tfID)
 	if err != nil {
-		if response.WasNotFound(resp.HttpResponse) {
-			d.SetId("")
-			return nil
-		}
-		return fmt.Errorf("making Read request on %s: %+v", *id, err)
+		return err
+	}
+	if js == nil {
+		d.SetId("")
+		return nil
 	}
 
+	id, err := jobschedule.ParseJobScheduleID(pointer.From(js.Id))
+	if err != nil {
+		return fmt.Errorf("parse jobSchedule id %s: %v", pointer.From(js.Id), err)
+	}
+
+	d.Set("resource_manager_id", js.Id)
```

Comment:
Can we ensure the id is case sensitive by doing:

```suggestion
	d.Set("resource_manager_id", id.ID())
```